### PR TITLE
Publication utils update

### LINKED ISSUE #46: fix publication doi in add_publication functions
**Describe the bug**
To conform to datacite/crossref doi display guidelines, doi urls should always be 'https://www.doi.org/{doi}' Currently the add_publication functions do not respect this.
**To Reproduce**
n/a

### PR DESCRIPTION
**Pull request by anngvu**
To test with curated spreadsheet:
```
# defaults to dry_run
new_pubs <- add_publications_from_file("Publications-NFOSI-202.tsv", publication_table_id, study_table_id) 
# view table
new_pubs$asDataFrame()
.syn$store(new_pubs)
```

Actual test copy with added entries [here](https://www.synapse.org/#!Synapse:syn32304014/tables/).

### CODE REVIEW

**Comment by anngvu** [comment]
The reason check fails is still because of 
```
❯ checking package dependencies ... WARNING
  Requires (indirectly) orphaned package: ‘gtools’ 
```

**Review by allaway — CHANGES_REQUESTED** [review]
Hey, so this worked on your test file but because the PMIDs had already been added to the table, it exited early (e.g. "PMID:... already exists in destination table"). 

I tried this with a PMID that is not in the table, and it worked on the dry_run=T, but when I tried to actually upload the results (dry_run=F), it errored out: 

```
> new_pubs <- add_publications_from_file("~/Downloads/Publications-NFOSI-202 - Sheet1.csv", "syn16857542", "syn16787123", dry_run = T) 
Rows: 123 Columns: 1                                                                                                                                                             
── Column specification ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
dbl (1): pmid

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

> new_pubs
<synapseclient.table.CsvFileTable>

> new_pubs$asDataFrame()
                          doi             diseaseFocus                                                     journal
1 https://www.doi.org/baac045 Neurofibromatosis type 1 Database : The Journal of Biological Databases and Curation
                                                                                            title year     pmid
1 Centralizing neurofibromatosis experimental tool knowledge with the NF Research Tools Database. 2022 35735230
                                                                                                                                                                                                                                                                                                 author
1 ["Ashley Clayton","Mialy DeFelice","Brynn Zalmanek","Jay Hodgson","Caroline Morin","Stockard Simon","Julie A Bletz","James A Eddy","Milen Nikolov","Jineta Banerjee","Kalyan Vinnakota","Marco Marasca","Kevin J Boske","Bruce Hoff","Ljubomir Bradic","YooRi Kim","James R Goss","Robert J Allaway"]
          manifestation fundingAgency         studyId
1 ["Hyperexcitability"]      ["NTAP"] ["syn11374345"]
                                                                                                                                                                          studyName
1 ["Leveraging human induced pluripotent stem cells (iPSCs) to determine the impact of patient-derived NF1 gene mutations on peripheral sensory neuron-driven Schwann cell growth"]

> new_pubs <- add_publications_from_file("~/Downloads/Publications-NFOSI-202 - Sheet1.csv", "syn16857542", "syn16787123", dry_run = F) 
Rows: 123 Columns: 1                                                                                                                                                             
── Column specification ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
dbl (1): pmid

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Error in py_get_attr_impl(x, name, silent) : 
  AttributeError: 'CsvFileTable' object has no attribute 'pmid'
  ```
  
  What do you think?

File: R/add_publication_from_pubmed.R [hunk 1]
```diff
@@ -1,100 +1,137 @@
...
+      # If batch mode, rbind and defer table schemafication until all records processed
+      if(batch) {
+        new_data <<- rbind(new_data, record)
+        if(counter == batch) new_data <- as_table_schema(new_data, publication_table_id) else return()
+      } else {
+        new_data <- as_table_schema(record, publication_table_id)
+      }
+      if(!dry_run) {
+        new_data <- .syn$store(new_data)
+        message(glue::glue('PMID:{new_data$pmid} added!'))
```
Comment (allaway) [inline R/add_publication_from_pubmed.R]: I think this is the culprit.

File: R/add_publication_from_pubmed.R [hunk 2]
```diff
@@ -1,100 +1,137 @@
...
+      
+      # If batch mode, rbind and defer table schemafication until all records processed
+      if(batch) {
+        new_data <<- rbind(new_data, record)
+        if(counter == batch) new_data <- as_table_schema(new_data, publication_table_id) else return()
+      } else {
+        new_data <- as_table_schema(record, publication_table_id)
+      }
+      if(!dry_run) {
+        new_data <- .syn$store(new_data)
```
Comment (allaway) [inline R/add_publication_from_pubmed.R:40]:
```suggestion
        new_data_stored <- .syn$store(new_data)
```

File: R/add_publication_from_pubmed.R [hunk 3]
```diff
@@ -1,100 +1,137 @@
...
+      
+      # If batch mode, rbind and defer table schemafication until all records processed
+      if(batch) {
+        new_data <<- rbind(new_data, record)
+        if(counter == batch) new_data <- as_table_schema(new_data, publication_table_id) else return()
+      } else {
+        new_data <- as_table_schema(record, publication_table_id)
+      }
+      if(!dry_run) {
+        new_data_stored <- .syn$store(new_data)
```
Comment (allaway) [inline R/add_publication_from_pubmed.R]:
```suggestion
        new_data <- .syn$store(new_data)
```

File: R/add_publication_from_pubmed.R [hunk 4]
```diff
@@ -1,100 +1,137 @@
...
+        message(glue::glue('PMID:{new_data$pmid} added!'))
```
Comment (allaway) [inline R/add_publication_from_pubmed.R]:
```suggestion
        message(glue::glue('PMID:{new_data$asDataFrame()$pmid} added!'))
```

**Review by allaway — APPROVED** [review]

**Comment by allaway** [comment]
Turns out it was an easy fix!

**Comment by anngvu** [comment]
@allaway Awesome, thanks for the fix!

**Pull request merged**